imports/plugins/included/marketplace/server/methods.js

```diff
@@ -1,23 +1,43 @@
-// import { Meteor } from "meteor/meteor";
-// // import { check } from "meteor/check";
-// // import { Reaction } from "/lib/api";
-// // import { Shops } from "/lib/collections";
-//
-//
-// Meteor.methods({
-//   // "marketplace/updateShopDetails": function (doc, _id) {
-//   //   check(_id, String);
-//   //   check(doc, Object);
-//   //
-//   //   if (!Reaction.hasPermission("admin", this.userId, Reaction.getSellerShopId(this.userId))) {
-//   //     return;
-//   //   }
-//   //
-//   //   Shops.update(_id, doc, function (error) {
-//   //     if (error) {
-//   //       throw new Meteor.Error(500, error.message);
-//   //     }
-//   //   });
-//   // }
-//   //
-// });
+import { Meteor } from "meteor/meteor";
+import { check } from "meteor/check";
+import { Reaction } from "/lib/api";
+import { Shops } from "/lib/collections";
+import { SHOP_WORKFLOW_STATUS_ACTIVE, SHOP_WORKFLOW_STATUS_DISABLED } from "../lib/constants";
+
+const status = [
+  SHOP_WORKFLOW_STATUS_ACTIVE,
+  SHOP_WORKFLOW_STATUS_DISABLED
+];
+
+export function marketplaceUpdateShopWorkflow(shopId, workflowStatus) {
+  check(shopId, String);
+  check(workflowStatus, String);
+
+  if (shopId === Reaction.getPrimaryShopId()) {
+    throw new Meteor.Error(403, "Cannot change shop status");
```

Comment:
We're no longer using error codes here, this should be `Meteor.Error("access-denied", "Cannot change shop status");